consensus/src/simplex/signing_scheme/mod.rs

```diff
@@ -0,0 +1,210 @@
+//! Signing-scheme abstraction and implementations used by `simplex`.
+//!
+//! The [`Scheme`] trait defines the cryptographic surface consumed by the core consensus engine:
+//! how votes are signed and verified, how quorum certificates are assembled/checked, and whether
+//! additional randomness is exposed for leader rotation. Two concrete schemes are provided:
+//!
+//! * [`bls12381_threshold`] – aggregated threshold signatures that also expose per-view randomness.
+//! * [`ed25519`] – quorum signatures collected into a vector.
+
+pub mod bls12381_threshold;
+pub mod ed25519;
+
+use crate::{
+    simplex::types::{Vote, VoteContext, VoteVerification},
+    types::Round,
+};
+use commonware_codec::{Codec, CodecFixed, Encode, Read};
+use commonware_cryptography::Digest;
+use commonware_utils::union;
+use rand::{CryptoRng, Rng};
+use std::{collections::BTreeSet, fmt::Debug, hash::Hash};
+
+/// Cryptographic surface required by `simplex`.
+///
+/// A `Scheme` produces validator votes, validates them (individually or in batches), assembles
+/// quorum certificates, checks recovered certificates and, when available, derives a randomness
+/// seed for leader rotation. Implementations may override the provided defaults to take advantage
+/// of scheme-specific batching strategies.
+pub trait Scheme: Clone + Debug + Send + Sync + 'static {
+    /// Vote signature emitted by individual validators.
+    type Signature: Clone + Debug + PartialEq + Eq + Hash + Send + Sync + CodecFixed<Cfg = ()>;
+    /// Quorum certificate recovered from a set of votes.
+    type Certificate: Clone + Debug + PartialEq + Eq + Hash + Send + Sync + Codec;
+    /// Randomness seed derived from a certificate, if the scheme supports it.
+    type Seed: Clone + Encode + Send;
+
+    /// Converts the scheme into a pure verifier.
+    ///
+    /// The returned instance should return `false` from `can_sign()`.
+    fn into_verifier(self) -> Self;
```

Comment:
Why do we need this? I would think `can_sign()` should be sufficient?

Reply:
It isn't strictly needed, just used for situations where we don't really need to sign, e.g. when you pass a scheme into marshal it makes sure to drop the signing material by calling this.

Reply:
I still kept it, feel free to remove if you don't think it's worth it.

Reply:
Removed it: https://github.com/commonwarexyz/monorepo/pull/1925